spec/unit/intercom/traits/api_resource_spec.rb

```diff
@@ -97,7 +112,7 @@
   it 'an initialized ApiResource is equal to one generated from a response' do
     class ConcreteApiResource; include Intercom::Traits::ApiResource; end
     initialized_api_resource = ConcreteApiResource.new(object_json)
-    except(object_json, 'type').keys.each do |attribute|
+    except(object_json, 'type', 'nested_fields').keys.each do |attribute|
```

Comment:
Temporarily added. Given there are two issues,
- Typed objects are created only when the hash has a key `type` as string. Symbol is not taken into account.
- The test checks for equality, The typed objects currently have no definition for equality.

Reply:
Equality for lightweight classes will be addressed in a separate PR.